index.js

```diff
@@ -0,0 +1,82 @@
+'use strict';
+/* eslint-disable no-console */
+
+class Datastore {
+    /**
+     * Constructor for Datastore
+     * @method constructor
+     * @param  {Object}    config Configuration
+     * @return {Datastore}
+     */
+    constructor(config) {
+        this.config = config;
+    }
+
+    /**
+     * Reload configuration
+     * @method configure
+     * @param  {Object}     config      Configuration
+     * @param  {Function}   [callback]
+     */
+    configure(config) {
+        this.config = config;
+    }
+
+    /**
+     * Get a single record given an id
+     * @method get
+     * @param  {Object}   config            Configuration
+     * @param  {String}   config.table      table name
+     * @param  {Object}   config.params     query params
+     * @param  {Object}   config.params.id  Unique id
+     * @param  {Function} callback
+     */
+    get() {
+        console.error('get is not implemented');
+        throw new Error('not implemented');
+    }
+
+    /**
+     * Save or create a record in the datastore
+     * @method save
+     * @param  {Object}   config                Configuration
+     * @param  {String}   config.table          table name
+     * @param  {Object}   config.params         record data
+     * @param  {Object}   [config.params.id]    Unique id
+     * @param  {Function} callback
+     */
+    save() {
+        console.error('save is not implemented');
+        throw new Error('not implemented');
+    }
+
+    /**
+     * Fetch multiple records from datastore
+     * @method scan
+     * @param  {Object}   config                Configuration
+     * @param  {String}   config.table          table name
+     * @param  {Object}   config.params         query parameters
+     * @param  {Object}   config.paginate       pagination parameters
```

Comment:
That would be at best a recommendation unless we implement validation and require implementers of the interface to implement a function called by this one.